processor/processor.go

```diff
@@ -1697,6 +1705,11 @@ func (proc *Handle) processJobsForDest(partition string, subJobs subJob) *transf
 				&shallowEventCopy,
 				&backendconfig.DestinationT{},
 			)
+			if len(destinationID) != 0 {
+				enhanceWithDestination(&shallowEventCopy, &backendconfig.DestinationT{
+					ID: destinationID,
+				})
+			}
```

Comment:
I fear this won't be reliable under all circumstances, at least if there are mixed events for the same source in the same batch, i.e. some events having a destination_id and some others not having anything or having another destination_id: Both `TransformerEvent.Metadata` & `TransformerEvent.Destination` are being overwritten during `Handle#validateEvents` based on the value found in the first event of the batch! (see `proc#getTransformerEvents`)

We could fill `TransformerEvent.Metadata.DestinationID` and then read it from `TransformerResponse.Metadata.DestinationID`, but having to rely on an external service for making sure that this field is being transferred properly is not the safest approach. Furthermore, `proc#getTransformerEvents` is a generic method used by other steps as well & we wouldn't want to break other steps :/

We might need to find an alternative strategy

Reply:
Ack have thought of another strategy will update the PR